### PR TITLE
Dockerfile dependencies update and multi-arch support

### PR DESCRIPTION
**Pull request by richerve**
## Goals and expectations

- To remove the hardcoded amd64 compilation of the go pkg so the image can be compiled for arm64 too
- To update the Go version

## Changes

Update Dockerfile

- Bump golang version
- Bump helm version
- Remove hardcoded ARCH in go build
- Minor cosmetic changes

Both the golang and helm images are multi-arch so they will build, or package, the proper binary when running in arm64 or amd64

I'm thinking on fully removing the drone.yml file as the build and publish will happen out of this repo.

## Local testing

Run the drone-helm3 plugin locally with podman, or docker, to install a helm chart.

```sh
MYAPISERVER="someurl"
MYNAMESPACE="somenamespace"
MYTOKEN="kubetoken"

podman build -t drone-helm3:multi-arch
podman run --rm -e PLUGIN_MODE=upgrade -ePLUGIN_ADD_REPOS="bitnami=https://charts.bitnami.com/bitnami" -e PLUGIN_CHART=bitnami/nginx -ePLUGIN_KUBE_API_SERVER=${MYAPISERVER} -ePLUGIN_KUBE_TOKEN=${MYTOKEN} -ePLUGIN_NAMESPACE=${MYNAMESPACE} -ePLUGIN_RELEASE=drone-nginx-example localhost/drone-helm3:multi-arch
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,15 @@
-# --- Build the plugin cli first ---
-FROM golang:1.18-alpine as builder
+FROM golang:1.21 as build
```
Comment (dskatz) [inline Dockerfile:1]: The golang version shouldn't be bumped without also bumping the `go.mod`.

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
+COPY go.mod go.sum ./
 RUN go mod download
 COPY . .
-
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /go/bin/drone-helm ./cmd/drone-helm
+RUN CGO_ENABLED=0 go build -o /go/bin/app ./cmd/drone-helm
 
 # --- Copy the cli to an image with helm already installed ---
-FROM alpine/helm:3.8.1
+FROM alpine/helm:3.13.3
```
Comment (dskatz) [inline Dockerfile]: I do not think we should move the helm version in this change.

**Review by dskatz — COMMENTED** [review]
I think this change should be limited to just supporting multi-arch.

File: Dockerfile (hunk 2 above)
Comment (richerve) [reply Dockerfile]: any particular reason?

Comment (dskatz) [reply Dockerfile]: Limits the scope of change and what needs to be tested.  If we are upgrading go.mod and helm, we need to test and evaluate that. 

If this PR is just focused on making this image available in arm64, then what we need to test is smaller

**Comment by whwalter** [comment]
Do you have details for what this change is trying to accomplish and how the changes should be tested locally?

File: Dockerfile (hunk 2 above)
Comment (dskatz) [reply Dockerfile]: Just note, I am currently going to be reading all the release notes between 3.8.1 and 3.13.3

Comment (richerve) [reply Dockerfile]: Reverted the change to use 3.8.1

**Review by richerve — COMMENTED** [review]
> Do you have details for what this change is trying to accomplish and how the changes should be tested locally?  

Added more context in the description

**Comment by dskatz** [comment]
I need to adjust the command in order to test locally in a kind cluster

- Set `--net=host` in order to reference the local API server
- Add `-ePLUGIN_SKIP_TLS_VERIFY=true`
- override the service.type ` -ePLUGIN_STRING_VALUES="service.type=ClusterIP"`


```
podman run --net=host --rm -e PLUGIN_MODE=upgrade -ePLUGIN_ADD_REPOS="bitnami=https://charts.bitnami.com/bitnami" -e PLUGIN_CHART=bitnami/nginx -ePLUGIN_KUBE_API_SERVER=${MYAPISERVER} -ePLUGIN_KUBE_TOKEN=${MYTOKEN} -ePLUGIN_NAMESPACE=${MYNAMESPACE} -ePLUGIN_SKIP_TLS_VERIFY=true -ePLUGIN_STRING_VALUES="service.type=ClusterIP" -ePLUGIN_RELEASE=drone-nginx-example localhost/drone-helm3
```

**Review by dskatz — COMMENTED** [review]
Looks good and local testing is working with some minor adjustments.

The `.drone.yaml` file needs several changes
 - golanglint-ci image version tag needs to be updated to support golang 1.21
 - same for the snyk step
 - The signature needs an update.  The build is still pending.

**Review by dskatz — APPROVED** [review]

**Pull request merged**